### ISSUE TITLE
Add network selection to new wallet+node flow

### ISSUE THREAD

**Issue by avive** [spacemeshos/smapp#837]
## Motivation
We currently have a hack-ish but super-useful network selector which is displayed once in a new smapp session. We'd like to move this into the new wallet creation flow and have a proper visual design for it.

## Tasks

1. Add new network selection screen new wallet creation flow - not only in the wallet-only mode, but also when creating a wallet+local-node.
2. Use the screen when user selects to switch a wallet (and local node if applicable) to a new network.
3. Once the flows for wallet-only and for wallet+local-node are updated, don't display the network selection screen on startup.

## Visual Mocks
We already have the designs for these screens. They need to be modified a bit.

### Existing mocks

![connect1](https://user-images.githubusercontent.com/96002/146175795-0e67cf0c-d865-4fdf-b499-7b00bc5f1819.png)

![connect2](https://user-images.githubusercontent.com/96002/146175787-df5c641f-5927-42d3-b9d2-1aae8591c389.png)

- These mocks should be used when a user sets up a wallet-only mode wallet as we want to enable the user to connect to any public network api and to provide a custom url from community networks which provide a public api.

### Updated Mocks

![Select Spacemesh Network](https://user-images.githubusercontent.com/96002/146177074-81b80c5d-278a-4c39-953c-fb0fc2d18d80.png)

![Select Spacemesh Network-1](https://user-images.githubusercontent.com/96002/146177067-842f6cb8-fdfa-4c97-8a0a-5a66e485c937.png)

- The friendly name of all Spacemesh public networks (e.g. testnets and devents) should be listed in the drop-down.

- Clicking on a network in the list should open the network dashboard in the user's web-browser so user can preview it before joining.

- The selected network should be used for the user's wallet and for its local node. e.g. the local node should use that network's config file obtainable from the discovery service.

**Comment by brusherru** [comment]
@avive 
1. The first screen that a new user will see is still a "Welcome" screen, right? In this case, I will remove all the sentences which show up a network name? Or we can put this new screen before all (as it does the popup right now, but it will be stylized).
2. Let's think a little bit ahead. We will probably want to add the "switch network" button to the settings or to the network screen. In this case, I propose to untie the network selection from the wallet setup (remove the panel on the left). Because the choice of the network affects the node and the config that it uses, but in fact does not affect the wallet itself. And probably it will be a better user experience in case if User will need to switch the network. Does it sound reasonable to you?

To sum up, I propose to:
- remove any mentioning of network name on the Welcome screen
- the next step is this brand-new network selection screen
- and this screen n is not a part of the wallet setup (no left panel)
- then we can add a feature to switch the network (open network selection screen) from the settings screen

Do you agree?

**Comment by avive** [comment]
Thinking a bit about this. Good observations and ideas.

**Comment by avive** [comment]
We need to go over all the major wallet flows and get a feel of how it might work. I still think that it is a good thing to strongly associate a wallet with a specific network and mode (wallet-only / local node) so it is clear to the user. It makes sense to me to enable the switch mode and network for a wallet via the wallet settings in the setting screen as this is a relatively advanced feature. We already let users switch between wallet only and managed node for the current wallet.
Let's gather the requirements:
- Network selection should be part of the create wallet flow. User should specify which network he wants to work with from the available listed public devnets/testnets via the discover service.
- User should be able to switch a created wallet to be used on a new network.
- User should be able to switch between wallet-only-mode and managed node for a created wallet.

I don't think we should add a button to switch networks from the network screen so we keep it simple and read-only to display the status of the current network and user can switch wallet to a different network via settings at any time.

**Comment by brusherru** [comment]
Ok, sounds reasonable.

**Comment by avive** [comment]
Regarding the welcome screen - I think it is okay to remove mention to a specific network as the user will need to select a network as part of the create new wallet flow for their wallet. It makes sense.
